src/main/java/com/exasol/dbbuilder/dialects/exasol/ExasolSchema.java

```diff
@@ -39,6 +37,7 @@ public ExasolSchema(final ExasolImmediateDatabaseObjectWriter writer, final Iden
      */
     public AdapterScript createAdapterScript(final String name, final AdapterScript.Language language,
             final String content) {
+        verifyNotDeleted();
```

Comment:
Actually the current solution does not convince me to 100%.
It requires changes in many places and I think it is hard to verify, that no place has been forgotten. 
Also in the potential case when creating new types of database objects or when extending the existing classes there is the risk of creating new methods and forgetting to guard them with `verifyNotDeleted()`.

A potential solution that comes into my mind would be to change `AbstractImmediateDatabaseObjectWriter.writeToObject()` and maybe some additional places of that kind. 

On the other hand my proposal could lead to `DatabaseObjectDeletedException` being thrown at a later point in time, which might deviate from the fail-fast-principle. This especially is true in the current case for `AdapterScript.createAdapterScript()` as the method will only return a builder without immediate write operation to the database. 

Also I don't want to block the PR too long. 🙂 

I am happy to discuss.